src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { ActivityIndicator, SectionList, StyleSheet, View } from 'react-native'
+import SectionHead from 'src/components/SectionHead'
+import GetStarted from 'src/home/GetStarted'
+import { useSelector } from 'src/redux/hooks'
+import { getFeatureGate, getMultichainFeatures } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import colors from 'src/styles/colors'
+import { Spacing } from 'src/styles/styles'
+import NoActivity from 'src/transactions/NoActivity'
+import { useTransactionFeedV2Query } from 'src/transactions/api'
+import EarnFeedItem from 'src/transactions/feed/EarnFeedItem'
+import NftFeedItem from 'src/transactions/feed/NftFeedItem'
+import SwapFeedItem from 'src/transactions/feed/SwapFeedItem'
+import TokenApprovalFeedItem from 'src/transactions/feed/TokenApprovalFeedItem'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import { pendingStandbyTransactionsSelector } from 'src/transactions/reducer'
+import { NetworkId, type TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type PaginatedData = {
+  [timestamp: number]: TokenTransaction[]
+}
+
+// Query poll interval
+const POLL_INTERVAL = 10000 // 10 secs
+const FIRST_PAGE_TIMESTAMP = 0
+
+function getAllowedNetworkIdsForTransfers() {
+  return getMultichainFeatures().showTransfers.join(',').split(',') as NetworkId[]
+}
+
+function renderItem({ item: tx }: { item: TokenTransaction; index: number }) {
+  switch (tx.__typename) {
```

Comment:
Could we use `tx.type` instead?
`__typename` is something added by GraphQL and I wanted to avoid relying on it with the new implementation.